lib/src/linting/rules/no-primitives-in-request.ts

```diff
@@ -0,0 +1,32 @@
+import { Contract } from "../../definitions";
+import { LintingRuleViolation } from "../rule";
+import { dereferenceType, TypeTable } from "../../types";
+
+/**
+ * Request types should always be object types
+ *
+ * @param contract a contract
+ */
+export function noPrimitivesInRequest(
+  contract: Contract
+): LintingRuleViolation[] {
+  const violations: LintingRuleViolation[] = [];
+  const typeTable = TypeTable.fromArray(contract.types);
+
+  contract.endpoints.forEach(endpoint => {
+    const { request } = endpoint;
+    const body = request && request.body;
+    if (!body) {
+      return;
+    }
+    const bodyType = dereferenceType(body.type, typeTable);
+
+    if (bodyType.kind !== "object") {
```

Comment:
For type checks, we use the type guards defined in https://github.com/airtasker/spot/blob/master/lib/src/types.ts#L353 - in this case `isObjectType(...`